File: C#/303_Range Sum Query - Immutable.cs
using System.Collections.Generic;
public class NumArray {
    
    List <int> sum = new List<int>();
    int [] nums;
    public NumArray(int[] nums) {
        this.nums = nums;
        sum.Add(0);
        for (int i = 0;i<nums.Count();i++)
        {            
           sum.Add(nums[i] + sum[i]);            
        }
        
    }
    
    public int SumRange(int i, int j) {
        return sum[j+1] - sum[i];
    }
}

File: Python/303_Range Sum Query - Immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.nums =nums
        self.sum = [0]
        for index,n in enumerate(self.nums):
            if index == 0 :
                self.sum.append(n)
            else:
                self.sum.append(n + self.sum[-1])
        
        

    def sumRange(self, i: int, j: int) -> int:
        return  self.sum [j+1] - self.sum[i]